### PR TITLE
Add test, small refactor

### PR DESCRIPTION
**Pull request by sheppardv**
Hey, I'm mining xsh and Web Developer. 

I suggest to add some tests and small refactor. If you are good with this, I can work on some deep refactor and more tests.

Run `npm i; npm test`

### CODE REVIEW

File: lib/shield.js [hunk 1]
```diff
@@ -1,195 +1,185 @@
...
+      for (let key in k) {
+        if (k.hasOwnProperty(key))
+          this.set(key, k[key]);
+      }
+      return;
+    }
 
-        return this;
-    },
+    const opts = this.opts;
```
Comment (ShieldCoin) [inline lib/shield.js]: doesn't thins conflict with line 146 and 154?

**Review by ShieldCoin — CHANGES_REQUESTED** [review]
Can we delete .editorconfig as it is not really needed?

**Review by ShieldCoin — APPROVED** [review]

**Comment by sheppardv** [comment]
`.editrorconfig` is common config for all editors. It's pretty nice when multiple people work on same code, you don't need to argue about indentation and so on. It's automatically parsed with most editors.

**Comment by ShieldCoin** [comment]
Ahh I see, I'll revert my change

**Pull request merged**